### PR TITLE
arch/nrf53: add TIM support

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary
Ported from NRF52

- arch/nrf53: add TIM support

- boards/nrf5340-dk: add TIMER configuration

## Impact
New functionality

## Testing

nrf5340-dk/timer example on the app core

### CODE REVIEW

File: arch/arm/src/nrf53/nrf53_tim.c [hunk 1]
```diff
@@ -0,0 +1,857 @@
...
+  putreg32(value, ((struct nrf53_tim_priv_s *)dev)->base + offset);
+}
+
+/****************************************************************************
+ * Name: nrf53_tim_irq2reg
+ *
+ * Description:
+ *   Get the value of the interrupt register corresponding to the given
+ *   interrupt source
+ *
```
Comment (acassis) [inline arch/arm/src/nrf53/nrf53_tim.c:222]: Please include the Input Parameters and the Return

File: arch/arm/src/nrf53/nrf53_tim.c [hunk 2]
```diff
@@ -0,0 +1,857 @@
...
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nrf53_tim_getreg
+ *
+ * Description:
+ *   Get a 32-bit register value by offset
+ *
```
Comment (acassis) [inline arch/arm/src/nrf53/nrf53_tim.c:188]: Please include the Input Parameters and the Return

File: arch/arm/src/nrf53/nrf53_tim.c [hunk 3]
```diff
@@ -0,0 +1,857 @@
...
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nrf53_tim_init
+ *
+ * Description:
+ *   Initialize TIMER device
+ *
```
Comment (acassis) [inline arch/arm/src/nrf53/nrf53_tim.c:769]: Please include the Input Parameters and the Return

File: arch/arm/src/nrf53/nrf53_tim.c (hunk 1 above)
Comment (raiden00pl) [reply arch/arm/src/nrf53/nrf53_tim.c:222]: I prefer to keep it as it is for now, for compliance with NRF52

Comment (acassis) [reply arch/arm/src/nrf53/nrf53_tim.c:222]: Ok, hahaha

**Pull request merged**